federatedscope/core/auxiliaries/utils.py

```diff
@@ -60,14 +58,15 @@ def setup_logger(cfg):
 
     # if exist, make directory with given name and time
     if os.path.isdir(cfg.outdir) and os.path.exists(cfg.outdir):
-        cfg.outdir = os.path.join(cfg.outdir, "sub_exp" +
-                                  datetime.now().strftime('_%Y%m%d%H%M%S')
-                                  )  # e.g., sub_exp_20220411030524
+        outdir = os.path.join(cfg.outdir, "sub_exp" +
+                              datetime.now().strftime('_%Y%m%d%H%M%S')
+                              )  # e.g., sub_exp_20220411030524
         while os.path.exists(cfg.outdir):
             time.sleep(1)
-            cfg.outdir = os.path.join(
+            outdir = os.path.join(
                 cfg.outdir,
                 "sub_exp" + datetime.now().strftime('_%Y%m%d%H%M%S'))
+        cfg.outdir = outdir
```

Comment:
It seems like `cfg.outdir` is initialized here by `exp/sub_exp__%Y%m%d%H%M%S`. Please make sure the config file is stored in the same dir (cfg is stored in the `freeze` function in config.py).

Reply:
We will call the logger_setup at the beginning and the following usages are based on the same outdir, see the new [PR](https://github.com/alibaba/FederatedScope/pull/30/files#diff-eb363534ce61d0fc725e670c047ab83e5358698478d3065641d4b0b4445baff6).